src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,470 @@
+) => {
+  if (Number.isInteger(step)) {
+    return value.toFixed(0);
+  }
+  return value.toFixed(numDecimals);
+};
+
+/**
+ * Properties for Slider component
+ */
+export type SliderProps = {
+  /**
+   * Set focus on first thumb in slider element.
+   * @default false
+   */
+  setFocus?: boolean;
+  /**
+   * Minimum slider value.
+   * @default 0
+   */
+  min?: number;
+  /**
+   * Maximum slider value.
+   * @default 100
+   */
+  max?: number;
+  /**
+   * Array of one or more values to show.
+   */
+  values: number[];
+  /**
+   *  Determines which segments are shown with color.
+   * 'none' - no colored tracks are displayed.
+   * 'auto' - no colored tracks are displayed.
+   * 'odd-segments'- colored tracks shown in segments 1,3,5, etc.
+   *                 Default if number of thumbs values are even.
+   * 'even-segments'- colored tracks shown in segments 0,2,4, etc.
+   *                 Default if number of thumbs values are odd.
+   * @default auto
+   */
+  trackDisplayMode?: TrackDisplayMode;
+  /**
+   * Step increment controls what values are allowed and the amount the value will
+   * change when left and right arrows are pressed when a Thumb has focus.
+   * @default 1
+   */
+  step?: number;
+  /**
+   * Forces control to be displayed in a disabled state where no interactive value
+   * changes are allowed.
+   *  @default false
+   */
+  disabled?: boolean;
+  /**
+   * Props to override default for showing a tooltip when Thumb is active or has focus.
+   * changes are allowed.
+   */
+  tooltipProps?: Partial<Omit<TooltipProps, 'content' | 'children'>>;
+  /**
+   * Optional tooltip function that can be supplied to generate tooltip text.
+   *  @default undefined
+   */
+  tooltipRender?: (val: number, step: number) => React.ReactNode;
+  /**
+   * Preferred position of tooltip relative to thumb.
+   * @default 'top'
+   */
+  /**
+   *  Optional array of labels that will be used to determine number of ticks
+   * displayed and their labels. Ticks are spaced evenly across width of Slider.
+   * @default undefined
+   */
+  tickLabels?: (string | JSX.Element)[];
+  /**
+   * Optional label for the minimum value. If undefined then the min
+   * value is shown. Use empty string for no label.
+   * @default undefined
+   */
+  minLabel?: string | JSX.Element;
+  /**
+   * Optional label for the maximum value. If undefined then the max
+   *  value is shown. Use empty string for no label.
+   * @default undefined
+   */
+  maxLabel?: string | JSX.Element;
+  /**
+   * Defines the allowed behavior when moving Thumbs when multiple Thumbs are
+   * shown.  Possible values:
+   * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+   * 'inhibit-crossing'- keeps the thumb from crossing and separated by a step.
+   * @default 'inhibit-crossing'
+   */
+  thumbMode?: ThumbMoveMode;
+  /**
+   * Additional props to include on slider container.
+   */
+  containerProps?: Omit<CommonProps, 'title'>;
```

Comment:
Could just make this part of the SliderProps

Reply:
@veekeys - didn't you prefer this method? I am not opposed to adding it to SliderProps if Vykintas is OK with it.

Reply:
Not sure, what you mean @mayank99. There is 2 things here, which we might want to expose to use:
Thumb and the container. So having a prop for the container stuff, seemed logical..no?

Reply:
As a user, if I do `<Slider className='my-class' />`, I expect it to be applied on the container, not the thumb. Better than doing `<Slider containerProps={{ className: 'my-class' }} />`

Reply:
not gonna judge your expectations here, but..
I wanted to make same as all our input components..
Now I see we actually use `className` on container and we have `inputClassName` for the input..
So I guess lets follow same approach here too (kinda input here..).

Reply:
So do I add an inputClassName, and if so do all thumbs get that className?

Reply:
There is no input here, but it might make sense to do `thumbProps`.

But `containerProps` just does not make sense if it's applied on the root.

Reply:
SliderProps now include old `containerProps` and added `thumbProps` so thumbs could be targeted.